db-engine/src/main/java/com/splicemachine/db/impl/sql/GenericStatement.java

```diff
@@ -857,12 +881,66 @@ private void fourPhasePrepare(LanguageConnectionContext lcc,
         }
     }
 
+    private void handleExplainNode(LanguageConnectionContext lcc,
+                                   CompilerContext cc,
+                                   StatementNode statementNode,
+                                   boolean cacheMe) throws StandardException {
+        if (!(statementNode instanceof ExplainNode) || ((ExplainNode)statementNode).isSparkExplain()) {
+            return;
+        }
+
+        ExplainNode explainNode = (ExplainNode)statementNode;
+
+        String explained = getExplainedStatement(statementText);
+
+        GenericStatement queryNode = new GenericStatement(compilationSchema,explained, isForReadOnly /*this could be incorrect with updatableCursors*/, lcc);
+        queryNode.sessionPropertyValues = lcc.getCurrentSessionPropertyDelimited();
+
+        if(cacheMe) {
+            // similar to ANALYSE statement, EXPLAIN statement will
+            // force invalidate the currently cached statement if it exists
+            // the assumption here is when running EXPLAIN, the user expects
+            // the plan to correspond to what the system currently thinks is
+            // the best plan, retrieving it from the cache instead could be
+            // misleading to say the least.
+            lcc.removeStatement(queryNode);
+
+            // add the statement to the statement cache
+            lcc.lookupStatement(queryNode);
+        }
```

Comment:
This needs to be discussed, I'm slightly in favor of reusing the already cached statement if there's one.

Reply:
I added a comment on the corresponding JIRA ticket listing all the assumptions and asking for feedback, I will adapt it to whatever the consensus on the right behavior will be.